test/schema/conditional_workflow.yml

```diff
@@ -0,0 +1,38 @@
+apiVersion: beehive/v1
+kind: Workflow
+metadata:
+  name: beehive-deployment
+  labels:
+    app: example2
+spec:
+  strategy:
+    type: sequence
+  template:
+    metadata:
+      name: beehive-deployment
+      labels:
+        app: example
+        use-case: test
+  agents:
+    - agent1
+    - agent2
+    - agent3
+  exception: step3
+  steps:
+    - name: step1
+      agent: agent1
+      condition:
+        - if: '{$in.some_condition == True}'
+          then: step2
+          else: step3
+    - name: step2
+      agent: agent2
+    - name: step3
+      agent: agent3
+      condition:
+        - case: '{$in.some_condition == True}'
+          do: step2
+        - case: '{$in.some_condition == False}'
+          do: step3
+        - default:
+            do: step3
```

Comment:
Does this mean the step3 executed repeatedly?  How is this end the workflow execution?

Reply:
default should be what happens if none of the case conditions are met. so say if it was `condition == 3 do X; condition == 4 do Y; default do Z` ... with true/false probably would just use if/then

Reply:
In addition to the step name, how about allowing `exit` or `end` so like `do:exit`?

Reply:
> Does this mean the step3 executed repeatedly? How does this end the workflow execution?

yes :) I did this on purpose since we need to have iterations (for or while loop) in next iteration :)

Reply:
> In addition to the step name, how about allowing `exit` or `end` so like `do:exit`?

We will need to have loops.

Reply:
So is this workflow keep running?

Reply:
I’ll rewrite the example to allow it to end workflow but like any language you can have infinite loop.